src/self_encryptor.rs

```diff
@@ -343,11 +352,15 @@ where
                 continue;
             }
             state.chunks[i].in_sequencer = true;
-            let pos = get_start_end_positions(state.file_size, i).0;
-            let vec = decrypt_chunk(&mut *state, i).await?;
-            decrypted_chunks.push((vec, pos));
+            positions.push(get_start_end_positions(state.file_size, i).0);
+            decryption_futures.push(decrypt_chunk(&mut *state, i).await);
         }
     }
+    let decrypted_data = join_all(decryption_futures).await;
+    let mut pos_iter = positions.into_iter();
+    for chunk in decrypted_data {
+        decrypted_chunks.push((chunk?, pos_iter.next().unwrap_or(0)))
```

Comment:
`any` of failed decryption will cause `all left over` decrypted chunk to be dropped ?